kohttp/src/test/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilderTest.kt

```diff
@@ -120,4 +124,22 @@ class ClientBuilderTest {
             }
     }
 
+    @Test
+    fun `client with ssl material`() {
+        val sslFactory = SSLFactory.builder()
+                .withIdentity("keystores/identity.jks", "secret".toCharArray())
+                .withTrustStore("keystores/truststore.jks", "secret".toCharArray())
+                .build()
+
+        assertThatCode {
+            client {
+                sslConfig = SslConfig(
+                        sslSocketFactory = sslFactory.sslContext.socketFactory,
+                        trustManager = sslFactory.trustManager
+                )
+                hostnameVerifier = sslFactory.hostnameVerifier
```

Comment:
I'm happy to go with this configuration!

The general question here: 
In addition to `socketFactory` and `trustManager` we also have `hostnameVerifier`, `followSslRedirects`, `certificatePinner`, m/b something else - from user perspective should we also have them in `SslConfig`?

Reply:
Glad to hear you like the configuration up till now. 😄

If we indeed look again to the already existing properties we can move the properties you mentioned into the SslConfig class to group them together. I refactored it, but I am not quite sure if I refactored it correctly to the current state as my kotlin experience is low.

So what I did is I marked all members of SslConfig as optional by making them nullable. If i didn't do this I needed to provide a default value for all those values and I think it won't be very good to provide default values. We could provide default values for hostnameVerifier and followSslRedirects but the other I would prefer to keep nullable. What do you think?

The ClientBuilderImpl will add those properties if sslConfig has been set and if the internal properties are present. The sslConfig setter within the ClientBuilderImpl looks a bit verbose, but I couldn't come up with a better/more clever solution. Let me know if you have any tips!